refactor!: dropping PXE JSON RPC

benesjan opened the pull request:
PXE is becoming just a lib used by the wallet and for this reason I am dropping the corresponding JSON RPC Server.

This leads to a large refactor of basically all the setups that dealt with PXE and wallets.

This is a followup work of [this PR](https://github.com/AztecProtocol/aztec-packages/pull/17163) in which I had this tasks planned:
1. Drop PXE JSON RPC server,
2. have Wallet instantiate PXE instead of having it passed in as an arg to a constructor,
3. revisit the PXE methods exposed on the TestWallet.

In this PR I tackled points 1 and 2.

benesjan commented:
* **#17228** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17228?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17228?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

benesjan commented on yarn-project/aztec.js/src/wallet/wallet.ts:
```diff
@@ -191,6 +192,18 @@ const MessageHashOrIntentSchema = z.union([
   }),
 ]);
 
+const ContractMetadataSchema = z.object({
+  contractInstance: z.union([ContractInstanceWithAddressSchema, z.undefined()]),
+  isContractInitialized: z.boolean(),
+  isContractPublished: z.boolean(),
+}) satisfies ZodFor<ContractMetadata>;
+
+const ContractClassMetadataSchema = z.object({
+  contractClass: z.union([ContractClassWithIdSchema, z.undefined()]),
+  isContractClassPubliclyRegistered: z.boolean(),
+  artifact: z.union([ContractArtifactSchema, z.undefined()]),
+}) satisfies ZodFor<ContractClassMetadata>;
```
This was moved here from PXE as the PXESchema got dropped.

benesjan commented on yarn-project/aztec/src/bin/index.ts:
```diff
@@ -54,7 +53,6 @@ async function main() {
   program = injectAztecNodeCommands(program, userLog, debugLogger);
   program = injectMiscCommands(program, userLog);
   program = injectDevnetCommands(program, userLog, debugLogger);
-  program = injectWalletCommands(program, userLog, debugLogger);
```
Agreed with Grego in DMs that it makes sense to drop the wallet commands from `aztec` command. Having it on `aztec` was just a tech debt.

benesjan commented on yarn-project/accounts/src/testing/index.ts:
```diff
@@ -43,18 +43,6 @@ export function getInitialTestAccountsData(): Promise<InitialAccountData[]> {
- * @param testWalletOrPxe - Test wallet or pxe instance to use to get the registered accounts.
- * @returns A set of key data for each of the initial accounts.
- */
-export async function getDeployedTestAccounts(testWalletOrPxe: { getPxe(): PXE } | PXE): Promise<InitialAccountData[]> {
```
I replaced this with a `registerInitialSandboxAccountsInWallet` util that resides in the `end-to-end` package.

Thunkar replied:
I think I'd be fine with that utility living in the `test-wallet` package (people will connect the testwallet to sandbox)

benesjan commented on yarn-project/cli-wallet/src/utils/wallet.ts:
```diff
@@ -208,4 +223,23 @@ export class CLIWallet extends BaseWallet {
     return simulationResults;
   }
+
+  // Recently added when having the wallet instantiate PXE as PXE is now hidden. This forced me to expose this when
+  // refactoring `checkTx` cli-wallet command.
+  getContracts(): Promise<AztecAddress[]> {
+    return this.pxe.getContracts();
+  }
+
+  // Recently added when having the wallet instantiate PXE as PXE is now hidden. This forced me to expose this when
+  // refactoring `checkTx` cli-wallet command.
+  getNotes(filter: NotesFilter): Promise<UniqueNote[]> {
+    return this.pxe.getNotes(filter);
+  }
+
+  // Recently added when having the wallet instantiate PXE as PXE is now hidden. This forced me to expose this when
+  // refactoring `bridge_fee_juice` cli-wallet command.
+  // TODO: This should most likely just be refactored to getProtocolContractAddresses.
+  getPXEInfo(): Promise<PXEInfo> {
+    return this.pxe.getPXEInfo();
+  }
```
WDYT about these commands? Needed to expose them as we had the corresponding commands exposed on the CLI (getNotes was required by inspectTx and it prints out notes emitted in the tx).

As a dev being able to print out contracts and notes when debugging seems pretty useful.

Thunkar replied:
This makes sense in the CLIWallet, since it's just what it does. They do not belong in the public interface though.

benesjan commented on yarn-project/cli-wallet/README.md:
```diff
@@ -1,2 +1,32 @@
 # Aztec wallet Documentation
 
+### Run tests locally
```
Grego sent me this through DMs so I made AI generate readme from it.

benesjan commented on yarn-project/cli/src/utils/inspect.ts:
```diff
@@ -31,21 +26,19 @@ export async function inspectBlock(
   }
 }
 
 export async function inspectTx(
```
I copied the unpruned version to the cli-wallet. It became less useful here without the private info.

benesjan commented on yarn-project/end-to-end/src/bench/client_flows/account_deployments.test.ts:
```diff
@@ -1,7 +1,7 @@
 import { AztecAddress, type DeployOptions, Fr, type Wallet, publishContractClass } from '@aztec/aztec.js';
 import type { SponsoredFPCContract } from '@aztec/noir-contracts.js/SponsoredFPC';
-import type { TestWallet } from '@aztec/test-wallet';
+import type { TestWallet } from '@aztec/test-wallet/server';
```
In some e2e tests I started getting this error:

<img width="1622" height="690" alt="Image" src="https://github.com/user-attachments/assets/4a1f17d6-bdb1-4b87-adb2-f24af87a4a90" />

so I just decided to have all the e2e tests use the newly introduced server version of TestWallet.

benesjan commented on yarn-project/aztec/src/sandbox/banana_fpc.ts:
```diff
@@ -62,22 +62,22 @@ export async function setupBananaFPC(initialAccounts: InitialAccountData[], wall
 }
 
-export async function getDeployedBananaCoinAddress(wallet: TestWallet) {
+export async function registerDeployedBananaCoinInWalletAndGetAddress(wallet: TestWallet) {
```
using here TestWallet seems incorrect but we don't have `getContracts` on the `Wallet` interface. Shall we expose it there? (or smt. like `isContractRegistered`?)

Thunkar replied:
RegisterContract is a no-op if it's already registered

benesjan commented on yarn-project/aztec/src/sandbox/sponsored_fpc.ts:
```diff
@@ -13,11 +13,11 @@ export async function getSponsoredFPCAddress() {
 
-export async function getDeployedSponsoredFPCAddress(wallet: TestWallet) {
-  const fpc = await getSponsoredFPCAddress();
+export async function registerDeployedSponsoredFPCInWalletAndGetAddress(wallet: TestWallet) {
```
Same here. Wallet not currently complete enough for the interface to be usable here.

benesjan commented on yarn-project/end-to-end/src/e2e_p2p/gossip_network.test.ts:
```diff
@@ -97,7 +92,7 @@ describe('e2e_p2p_network', () => {
     // should be set so that the only way for rollups to be built
     // is if the txs are successfully gossiped around the nodes.
-    const contexts: NodeContext[] = [];
+    const txsSentViaDifferentNodes: SentTx[][] = [];
```
Here and in the tests that follow I dropped the NodeContext type:

```
export interface NodeContext {
  node: AztecNodeService;
  pxeService: PXEService;
  txs: SentTx[];
}
```

because only the txs from it were ever used.

benesjan commented on yarn-project/pxe/src/test/pxe_service.test.ts:
```diff
@@ -1,63 +1,146 @@
 
 import type { PXEServiceConfig } from '../config/index.js';
 import { PXEService } from '../pxe_service/pxe_service.js';
-import { pxeTestSuite } from './pxe_test_suite.js';
```
In this file I just merged the `pxe_test_suite` to it and moved the pxe service creation to the beforeAll block.

We had it done like this before because the `pxe_test_suite` was imported in e2e package.

benesjan commented on yarn-project/stdlib/src/interfaces/pxe.test.ts:
```diff
@@ -1,422 +0,0 @@
-import { L1_TO_L2_MSG_TREE_HEIGHT } from '@aztec/constants';
```
This was just testing the `PXESchema` and that schema no longer exists.

benesjan commented on yarn-project/stdlib/src/interfaces/pxe.ts:
```diff
@@ -241,6 +237,11 @@ export interface PXE {
   ): Promise<T[]>;
+
+  /**
+   * Stops the PXE's job queue.
```
Needed to expose this to be able to stop the pxe from wallet without needing to directly import the PXEService there. I think I would just merge the PXE interface with PXEService and rename that to PXE once this PR is merged.

benesjan commented on yarn-project/test-wallet/src/wallet/server.ts:
```diff
@@ -0,0 +1,87 @@
+ * Note that the only difference from `lazy` and `bundle` test wallets is that it uses the `createPXEService` function
+ * from the `pxe/server` package.
+ */
+export class TestWallet extends BaseTestWallet {
```
Needed to introduce this because I needed a version that uses the server version of `createPXEService` function.

benesjan commented on docker-compose.yml:
```diff
@@ -90,17 +61,16 @@ services:
     ports:
-      # the RPC endpoint - expose it on a different port to avoid conflicting with PXE
-      - 8999:8999/tcp
+      - 8080:8080/tcp
```
Now we don't need to worry about the port collisions so I just used the port 8080 here.

Thunkar commented on spartan/aztec-network/templates/_helpers.tpl:
```diff
@@ -60,10 +60,6 @@ imagePullPolicy: {{ .Values.images.aztec.pullPolicy }}
 
 
 
-{{- define "aztec-network.pxeUrl" -}}
```
Very scared of this, ping someone from the networks team!

benesjan replied:
Already let Mitch know about this 👍

Apparently all of this is anyway getting completely refactored soon so seems to be a good time to do this PR.

Thunkar commented on yarn-project/cli/src/utils/setup_contracts.ts:
```diff
@@ -17,12 +18,12 @@ export async function getSponsoredFPCAddress() {
     salt: new Fr(SPONSORED_FPC_SALT),
   });
-  await pxe.registerContract({ instance: sponsoredFPCInstance, artifact: SponsoredFPCContract.artifact });
+  await wallet.registerContract({ instance: sponsoredFPCInstance, artifact: SponsoredFPCContract.artifact });
```
This is useless for a short lived command, since the contract is just registered privately on a dead wallet. It might be useful to just print the address though.

benesjan replied:
Good catch!

Addressed in https://github.com/AztecProtocol/aztec-packages/pull/17228/commits/4a2b13148710d4496bcb1821bf00c0e4d04ac515

The sponsored fpc flag on the cli command was useless as we did not ever deploy the FPC there.

benesjan committed:
refactor!: dropping PXE JSON RPC

PXE is becoming just a lib used by the wallet and for this reason I am dropping the corresponding JSON RPC Server.

This leads to a large refactor of basically all the setups that dealt with PXE and wallets.

This is a followup work of [this PR](https://github.com/AztecProtocol/aztec-packages/pull/17163) in which I had this tasks planned:
1. Drop PXE JSON RPC server,
2. have Wallet instantiate PXE instead of having it passed in as an arg to a constructor,
3. revisit the PXE methods exposed on the TestWallet.

In this PR I tackled points 1 and 2.
